backends/dpdk/dpdkHelpers.cpp

```diff
@@ -768,26 +768,77 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
                         append_parser_name(parser, IR::ParserState::reject)));
             add_instr(new IR::DpdkLabelStatement(end_label));
         } else if (a->method->name == "add_entry") {
+            auto args = a->expr->arguments;
+            auto argSize = args->size();
+            const IR::Expression *timeout_id = nullptr;
+            if (argSize != 2 && argSize != 3) {
+                ::error(ErrorType::ERR_UNEXPECTED, "Unexpected number of arguments for %1%",
+                            a->method->name);
+                return false;
+            }
             auto action = a->expr->arguments->at(0)->expression;
             auto action_name = action->to<IR::StringLiteral>()->value;
             auto param = a->expr->arguments->at(1)->expression;
+            if (argSize == 3)
+                timeout_id = a->expr->arguments->at(2)->expression;
+            if (timeout_id->is<IR::Constant>() ||  argSize == 2) {
+                // FIXME This code is valid if specification defines timeout as optional param.
+                BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
+                IR::ID tmo(refmap->newName("timeout_id"));
+                if (argSize == 3)
```

Comment:
Can line 788 and line 794 if block be written under one if condition?

Reply:
Updated